atomic_reactor/dirs.py

```diff
@@ -135,6 +133,18 @@ def has_sources(self) -> bool:
                 return False
         return True
 
+    def init_build_dirs(self, platforms: List[str], source: Source) -> None:
+        """Initialize the root build directory with specific determined platforms.
+
+        :param list[str] platforms: a list of platforms to build for.
+        """
+        if not platforms:
+            raise ValueError("At least one platform must be specified.")
+        self.platforms = sorted(platforms)
+        if self.has_sources:
+            return
+        self.copy_sources(source)
```

Comment:
Should copy_sources be private now? It could be easy to misuse, if you call copy_sources() before setting platforms it will do nothing

Reply:
I mean everything here now can be misused without platforms

Reply:
That's a good point :disappointed:

Reply:
@rcerven Sorry, I can't get your point. How can everything be misused?

Reply:
I think the potential problem is that in the prebuild task, if you call `workflow.build_dir.for_each()` or `workflow.build_dir.for_all_copy()` before the "check_and_set_platforms" plugin did the initialization, the methods will silently do nothing. But maybe if we just run the check platforms plugin at the start of the prebuild task, that wouldn't be a big problem.

Or if we kill the plugin and instead determine the platforms in `PreBuildTask.execute` :-)

Reply:
I see. But, based on current plugins design, if a plugin calls either the for_each or for_all_copy before check_and_set_platforms plugin run, this case should be an error and not allowed. :)

Reply:
> Or if we kill the plugin and instead determine the platforms in PreBuildTask.execute :-)

I actually do think as well the platforms can be determined before the workflow starts the image build, and the determined platforms is an input for the image build, probably can be set into the user_params.

Reply:
> , probably can be set into the user_params.

Or, just as a workflow object attribute, e.g. `workflow.platforms`.